### PR TITLE
New: the `no-implicit-coercion` rule (#1621)

### PR DESCRIPTION
**Pull request by mysticatea**

### CODE REVIEW

**Comment by lo1tuma** [comment]
The `fixes` tag is missing in the commit message.

**Comment by mysticatea** [comment]
Oops, sorry.

**Comment by gyandeeps** [comment]
Your build is breaking because of this issue https://github.com/eslint/eslint/issues/3021

**Comment by ilyavolodin** [comment]
I restarted the build, should clear shortly

**Comment by gyandeeps** [comment]
I think we need a rebase.

**Comment by ilyavolodin** [comment]
Yeah, it doesn't look like I can restart AppVeyor builds. So Travis build should clear in a few minutes, but AppVeyor will still be failing.

**Comment by mysticatea** [comment]
OK, rebased!

File: docs/rules/no-implicit-coercion.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Disallow the type conversion with shorter notations. (no-implicit-coercion)
+
+On JavaScript, there are notations which are not understandable to beginners.
```
Comment (ilyavolodin) [inline docs/rules/no-implicit-coercion.md]: In JavaScript, there are a lot of different ways to convert variable types. Some of them might be hard to read and understand. (or something like that).

Comment (mysticatea) [inline docs/rules/no-implicit-coercion.md]: Thank you very much! I felt the description is very better.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**